plumbing/object/tree_test.go

```diff
@@ -262,6 +265,41 @@ func (s *TreeSuite) TestTreeWalkerNextNonRecursive(c *C) {
 	c.Assert(count, Equals, 8)
 }
 
+func (s *TreeSuite) TestTreeWalkerNextSubmodule(c *C) {
+	st, err := filesystem.NewStorage(fixtures.ByTag("submodule").One().DotGit())
```

Comment:
please, comment what repository are you using for this test, it is very hard to know what should be inside the commit below otherwise.